### ISSUE TITLE
False positive no-member for annotated Enum _value_

### ISSUE THREAD

**Issue by adamtheturtle** [PyCQA/pylint#11413]
### Bug description

The [typing specification](https://typing.python.org/en/latest/spec/enums.html#member-values) allows an enum class to annotate `_value_` so checkers can enforce and infer its member value type.

```python
import dataclasses
import datetime
import enum
from collections.abc import Callable

@dataclasses.dataclass(frozen=True)
class Payload:
    formatter: Callable[[datetime.date], str]

def format_date(value: datetime.date) -> str:
    return value.isoformat()

class FormatBase(enum.Enum):
    _value_: Payload

    def __call__(self, value: datetime.date) -> str:
        return self._value_.formatter(value)

class Format(FormatBase):
    _value_: Payload
    DEFAULT = Payload(formatter=format_date)
```

### Command and output

```console
$ pylint --disable=all --enable=no-member example.py
example.py:16:15: E1101: Instance of '_value_' has no 'formatter' member (no-member)
```

Mypy, Pyright, Pyrefly, and Ty accept the example without diagnostics.

### Expected behavior

No `no-member` diagnostic: `_value_` is explicitly declared as `Payload`.

### Versions

```text
pylint 4.0.8
astroid 4.0.4
Python 3.14.4
```
